functions/model/tasks/lib.js

```diff
@@ -1,47 +1,66 @@
-        CompletionDate: completiondate,
-        OrganizationId: orgId,
-        TeamId: teamId,
-        Type: type,
-    });
-    return Promise.resolve(createTask);
-};
-
-exports.updateTask = function(inputJson, orgDomain, taskId) {
-    const updateTaskPromise = db.collection("Organizations").doc(orgDomain).collection("Tasks").doc(taskId).update(inputJson);
-    return Promise.resolve(updateTaskPromise);
-};
-
-exports.getTask = function(taskId, orgDomain) {
-    const getTaskDetails = db.collection("Organizations").doc(orgDomain).collection("Tasks").doc(taskId).get().then((taskDoc) => {
-        return taskDoc.data();
-    });
-    return Promise.resolve(getTaskDetails);
-};
\ No newline at end of file
+/* eslint-disable linebreak-style */
+/* eslint-disable no-undef */
+/* eslint-disable object-curly-spacing */
+/* eslint-disable eol-last */
+/* eslint-disable indent */
+/* eslint-disable max-len */
+// const admin = require("firebase-admin");
+// const db = admin.firestore();
+
+const { db } = require("../application/lib");
+
+exports.setTask = function(orgDomain, taskId, title, des, priority, difficulty, creator, assignee, estimatedTime, status, project, loggedWorkTotalTime, workDone, sprintNumber, storyPointNumber, creationDate, completiondate, orgId, teamId, type) {
+    const createTask = db.collection("Organizations").doc(orgDomain).collection("Tasks").doc(taskId).set({
+        Id: taskId,
+        Title: title,
+        Description: des,
+        Priority: priority,
+        Difficulty: difficulty,
+        Creator: creator,
+        Assignee: assignee,
+        EstimatedTime: estimatedTime,
+        Status: status,
+        Project: project,
+        LogWorkTotalTime: loggedWorkTotalTime,
+        WorkDone: workDone,
+        SprintNumber: sprintNumber,
+        StoryPointNumber: storyPointNumber,
+        CreationDate: creationDate,
+        CompletionDate: completiondate,
+        OrganizationId: orgId,
+        TeamId: teamId,
+        Type: type,
+    });
+    return Promise.resolve(createTask);
+};
+
+exports.updateTask = function(inputJson, orgDomain, taskId) {
+    const updateTaskPromise = db.collection("Organizations").doc(orgDomain).collection("Tasks").doc(taskId).update(inputJson);
+    return Promise.resolve(updateTaskPromise);
+};
+
+exports.getTask = function(taskId, orgDomain) {
+    const getTaskDetails = db.collection("Organizations").doc(orgDomain).collection("Tasks").doc(taskId).get().then((taskDoc) => {
+        return taskDoc.data();
+    });
+    return Promise.resolve(getTaskDetails);
+};
+exports.getTasks = function(orgDomain, teamId, sprintNumber) {
+    const getTasks = db.collection("Organizations").doc(orgDomain).collection("Tasks").where("TeamId", "==", teamId).where("SprintNumber", "==", sprintNumber).get().then((task) => {
+        const tempDoc = task.docs.map((doc) => {
+            return doc.data();
+        })
+        return tempDoc;
+    });
+
+    return Promise.resolve(getTasks);
+};
```

Comment:
This method is directly performing the query, use the concept of default arguments and make the query dynamic so that based on data we will perform the query.

Making it much more accessible when needed next time with some different query

Reply:
The similar example is the method defined below.